src/core/actions/unitactions/Convert.java

```diff
@@ -22,19 +23,31 @@ public Unit getTarget() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute all the Convert actions that are possible
-        return null;
+        LinkedList<Action> converts = new LinkedList<>();
+        Board b = gs.getBoard();
+        if(isFeasible(gs)){
+            Convert a = new Convert(this.unit, this.target);
+            converts.add(a);
+        }
+        return converts;
     }
 
     @Override
     public boolean isFeasible(final GameState gs) {
-        //TODO: check if this Convert action is possible.
+        //Check if target in range
```

Comment:
We need to add a check for the tribe of the target unit (we're not converting units from our own tribe)